l1-contracts/contracts/state-transition/chain-deps/facets/Admin.sol

```diff
@@ -91,21 +92,36 @@ contract AdminFacet is Base, IAdmin {
                             UPGRADE EXECUTION
     //////////////////////////////////////////////////////////////*/
 
-    /// @notice Executes a proposed governor upgrade
+    /// @notice Executes an upgrade from the state transition
     /// @dev Only the current governor can execute the upgrade
     /// @param _diamondCut The diamond cut parameters to be executed
-    function executeUpgrade(Diamond.DiamondCutData calldata _diamondCut) external onlyGovernor {
+    function executeUpgrade(Diamond.DiamondCutData calldata _diamondCut) external onlyStateTransition {
         Diamond.diamondCut(_diamondCut);
         emit ExecuteUpgrade(_diamondCut);
     }
 
+    /// @notice Executes a proposed governor upgrade
+    /// @dev Only the current governor can execute the upgrade
+    /// @param _diamondCut The diamond cut parameters to be executed
+    /// @notice we have this function so the event can be emitted, 
+    /// @notice as the genesis upgrade is not a normal upgrade, so the server handles it differently  
+    /// @notice based on this event
+    function executeChainIdUpgrade(
```

Comment:
If we move the event there, then we will have to track the events differently in the server. I moved it here as the normal upgrades are emitted from here

Reply:
Why server can't read `_protocolVersion` from the initialization contract (genesisUpgrade) instead? Just think that the function is not needed as there is one for upgrades already

Reply:
its not the protocol version, its the event itself. It can probably read the event from there, but it was easier for me when modifiying the server to emit it here. I will now move it there, and tell Lyova about it so he can chage that, wdyt?

Reply:
Sounds good!

cc @ly0va

Reply:
Also, just to note, it will make filtering the tx a bit harder. So I will have to add the STC contract address by which the contract can be filtered.

Reply:
> Also, just to note, it will make filtering the tx a bit harder. So I will have to add the STC contract address by which the contract can be filtered.

Do you mean STM? I think it is fine, since Governance and STC already need to be filtered

But also we can move event to Diamond Initializor (`genesisUpgrade`), so then server don't need to filter STM events

Reply:
No, the STC contract address. Because the event is now in STM and not STC we don't know automatically which chain it corresponds to. So we can filter by STC address or chainId in the event, but we need to add one of them. The server has access to both of them from the env.